main/input_default.cpp

```diff
@@ -876,6 +876,10 @@ void InputDefault::joy_axis(int p_device, int p_axis, const JoyAxis &p_value) {
 
 	Joypad &joy = joy_names[p_device];
 
+	if (p_axis < 0 || p_axis >= JOY_AXIS_MAX) {
```

Comment:
I guess this should throw an error.
You can do it with `ERR_FAIL_INDEX(p_axis, JOY_AXIS_MAX)`